### DISCUSSION TITLE
Custom Extensions and Joomla API

### DISCUSSION THREAD

**Discussion by noxidsoft** [alexandreelise/j4x-api-collection#41] [Q&A]
1. Can I convert my custom extension(s) e.g. component, module, template, to expose new Joomla API end points?
2. What are the safety ramifications?
3. Is there current documentation to assist me?

**Comment by alexandreelise** [comment]
Hi @noxidsoft,  Alex here . Welcome on this discussion about Joomla 4.x Web Services.
I'll do my best to answer your questions one by one. Here it goes:

1 . You should be able to create your own custom component and integrate the newly custom made Web Services conforming to the new Joomla! 4.x standards of making such Web Services. For modules and templates, I think you could do it by creating a plugin of type webservices that exists since Joomla 4.0.0 out since august 17th 2021. Or you could use system plugin which is much more powerful that can hook into any event. Both plugin types can handle Joomla! 4.x Web Services custom routes you might have created.

2. I don't really understand what you mean by "safety ramifications" can you clarify?

3. Yes there is an official Joomla documentation about the new Joomla Core Web Services 
  - [Official Documentation ](https://docs.joomla.org/Category:Joomla!_4.x_Web_Services)
  - [This example component](https://github.com/alexandreelise/com_helloapi) I made to help people like you get up and running and help the community
